src/main/java/seedu/address/logic/commands/DeletePostalCodeCommand.java

```diff
@@ -0,0 +1,85 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+/**
+ * Deletes all customers with given postal code
+ */
+public class DeletePostalCodeCommand extends Command {
+
+    public static final String COMMAND_WORD = "deletePC";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes all customers with given postal code.\n"
+            + "Parameters: POSTAL CODE (must be a 6 digit code)\n"
+            + "Example: " + COMMAND_WORD + " 540123";
+
+    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted customer(s) with postal code %1$s: %2$s";
+
+    private final String postalCode;
+
+    public DeletePostalCodeCommand(String postalCode) {
+        this.postalCode = postalCode;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        if (!isValidPostalCode(postalCode)) {
+            throw new CommandException(String.format(Messages.MESSAGE_INVALID_POSTAL, postalCode));
+        }
+        List<Person> peopleToDelete = model.getPeopleByPostalCode(postalCode);
+
+        if (peopleToDelete.isEmpty()) {
+            throw new CommandException(Messages.MESSAGE_NO_MATCH);
+        }
+
+        for (Person personToDelete : peopleToDelete) {
+            model.deletePerson(personToDelete);
+        }
+
+        List<String> deletedNames = peopleToDelete.stream()
+                .map(person -> person.getName().toString())
+                .collect(Collectors.toList());
+        String namesString = String.join(", ", deletedNames);
+
+        return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, postalCode, namesString));
+    }
+    /**
+     * Validates the postal code format.
+     * @param postalCode The postal code to validate.
+     * @return true if the postal code is valid; false otherwise.
+     */
+    private boolean isValidPostalCode(String postalCode) {
```

Comment:
I believe `isValidPostalCode` should be a `public static` method under `PostalCode`